assets/blocks/course-outline/status-preview/use-course-lessons-for-preview-status.js

```diff
@@ -0,0 +1,36 @@
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { COURSE_STATUS_STORE } from './status-store';
+
+/**
+ * A hook to update the status store when a lesson is removed.
+ *
+ * @param {string}  clientId  The outline block id.
+ * @param {boolean} isPreview Whether the block is currently in preview mode.
+ */
+export const useCourseLessonsForPreviewStatus = function (
```

Comment:
Maybe rename this to somthing like `useLessonStatusSync`? 

I think that the `Preview` being in both the name and in an argument while being unrelated might cause confusion.